API/RecommendationService/scenario_service.py

```diff
@@ -0,0 +1,128 @@
+            {"name": "@cmd", "value": "az " + commands[-1]},
+        ] + [{"name": "@src"+str(int(src)), "value": src} for src in source_type],
+        enable_cross_partition_query=True,
+    ):
+        if len(item['commandSet']) > 1:
+            scenario = {
+                'scenario': item['name'],
+                'nextCommandSet': strip_az_in_command_set(item['commandSet'][1:]),
+                'source': RecommendationSource.OfflineCaculation,
+                'type': RecommendType.Scenario
+            }
+            if 'description' in item:
+                scenario['reason'] = item['description']
+            result.append(scenario)
+
+    if len(result) >= top_num:
+        return result[0: top_num]
+
+    return result
+
+
+def get_search_results(trigger_commands: List[str], top: int = 5):
+    """Search related sceanrios using cognitive search
+
+    Args:
+        trigger_commands (List[str]): list of commands used to search
+        top (int, optional): top num of returned results. Defaults to 5.
+
+    Returns:
+        list[dict]: searched scenarios
+    """
+    if len(trigger_commands) == 0:
+        return []
+    service_endpoint = os.environ["SCENARIO_SEARCH_SERVICE_ENDPOINT"]
+    search_client = SearchClient(endpoint=service_endpoint,
+                                 index_name=os.environ["SCENARIO_SEARCH_INDEX"],
+                                 credential=AzureKeyCredential(os.environ["SCENARIO_SEARCH_SERVICE_SEARCH_KEY"]))
+
+    search_statement = "(" + " OR ".join([f'"{cmd}"' for cmd in trigger_commands][:-1]) + ") AND " + f'"{trigger_commands[-1]}"'
+    search_statement = f'"{trigger_commands[-1]}" OR ({search_statement})'
+    results = search_client.search(
+        search_text=search_statement,
+        include_total_count=True,
+        search_fields=["commandSet/command"],
+        highlight_fields="commandSet/command",
+        top=top,
+        query_type='full')
+    results = list(results)
+    return results
+
+
+def get_scenario_recommendation_from_search(command_list, top_num=5):
+    """Recommend Scenarios that current context could be in
+
+    Args:
+        command_list (list[str]): commands used to trigger
+        top_num (int, optional): top num of recommended results. Defaults to 5.
+
+    Returns:
+        list[dict]: searched scenarios
+    """
+    if len(command_list) == 0:
+        return []
+    trigger_len = int(os.environ.get("ScenarioRecommendationTriggerLength", "3"))
+    trigger_commands = get_latest_cmd(command_list, trigger_len)
+    trigger_commands = [cmd[3:] if cmd.startswith("az ") else cmd for cmd in trigger_commands]
+    searched = get_search_results(trigger_commands, top_num)
+
+    results = []
+    for item in searched:
+        cmds = [(idx, cmd['command'][3:]) for idx, cmd in enumerate(item['commandSet']) if len(cmd) > 3]
+        exec_idx = [cmd[0] for cmd in cmds if cmd[1] not in trigger_commands]
+        if len(cmds) - 1 not in exec_idx:
+            continue
```

Comment:
I didn't understand the code here. Could we make it more readable here?

Reply:
I will add some comments here.

Reply:
Done